crates/rspack_plugin_javascript/src/ast/minify.rs

```diff
@@ -196,50 +189,53 @@ pub fn minify(
           }
           .expect("Invalid extractComments");
-          let mut source = ConcatSource::default();
+          let mut extracted_comments = vec![];
           // add all matched comments to source
-          {
-            let (l, t) = comments.borrow_all();
-
-            l.iter().for_each(|(_, vc)| {
-              vc.iter().for_each(|c| {
-                if reg.is_match(&c.text) {
-                  source.add(RawSource::from("/*"));
-                  source.add(RawSource::from(&*c.text));
-                  source.add(RawSource::from("*/"));
-                  source.add(RawSource::from("\n"));
-                }
-              });
+
+          let (leading_trivial, trailing_trivial) = comments.borrow_all();
+
+          leading_trivial.iter().for_each(|(_, comments)| {
+            comments.iter().for_each(|c| {
+              if reg.is_match(&c.text) {
+                extracted_comments.push(match c.kind {
+                  CommentKind::Line => {
+                    format!("// {}", c.text)
+                  }
+                  CommentKind::Block => {
+                    format!("/*{}*/", c.text)
+                  }
+                });
+              }
             });
-            t.iter().for_each(|(_, vc)| {
-              vc.iter().for_each(|c| {
-                if reg.is_match(&c.text) {
-                  source.add(RawSource::from("/*"));
-                  source.add(RawSource::from(&*c.text));
-                  source.add(RawSource::from("*/"));
-                  source.add(RawSource::from("\n"));
-                }
-              });
+          });
+          trailing_trivial.iter().for_each(|(_, comments)| {
+            comments.iter().for_each(|c| {
+              if reg.is_match(&c.text) {
+                extracted_comments.push(match c.kind {
+                  CommentKind::Line => {
+                    format!("// {}", c.text)
+                  }
+                  CommentKind::Block => {
+                    format!("/*{}*/", c.text)
+                  }
+                });
+              }
             });
-          }
+          });
+
           // if not matched comments, we don't need to emit .License.txt file
-          if source.size() > 0 {
+          if !extracted_comments.is_empty() {
             all_extract_comments
               .lock()
               .expect("all_extract_comments lock failed")
               .insert(
                 filename.to_string(),
                 ExtractedCommentsInfo {
-                  source: source.boxed(),
+                  source: RawSource::Source(extracted_comments.join("\n\n")).boxed(),
                   comments_file_name,
                 },
               );
           }
-          // if comments are extracted, we don't need to preserve them
-          comments = SingleThreadedComments::default();
-        } else {
-          // if comments are not extracted, then we minify them
-          minify_file_comments(&comments, preserve_comments);
```

Comment:
Minify codegen does not need comments information, dropping it is enough, it could reduce comments processing time.